android/src/com/mapswithme/maps/MwmBroadcastReceiver.java

```diff
@@ -0,0 +1,22 @@
+package com.mapswithme.maps;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+
+public abstract class MwmBroadcastReceiver extends BroadcastReceiver {
+
+    protected abstract void onReceiveInitialized(Context context, Intent intent);
+
+    @Override
+    public final void onReceive(Context context, Intent intent)
+    {
+        MwmApplication app = MwmApplication.from(context);
+        if (!app.arePlatformAndCoreInitialized())
+        {
+            app.initCore();
```

Comment:
You should check returned value from initcore method, there is case when we cannot create dirs on file system and show error dialog to the user on application start, otherwise the crash will happen.

```
if (!app.arePlatformAndCoreInitialized() && !app.initCore())
    return;
```